openedx/openedx-platform
# Set SWIFT_CONTENT_LENGTH_FROM_FD=False

Works around a bug with Django SWIFT storage uploads in edx-platform encountered since upgrading to `django-swift-storage==1.2.18` (cf https://github.com/edx/edx-platform/pull/17487).

Also adds `SWIFT_LAZY_CONNECT=True` to `cms.envs.openstack`, to balance with the lms settings.

**JIRA tickets**: OC-4133, [OSPR-2315](https://openedx.atlassian.net/browse/OSPR-2315)

**Dependencies**:  https://github.com/dennisv/django-storage-swift/pull/100

**Sandbox URL**: sandbox provisioning

* LMS: https://pr17748.sandbox.opencraft.hosting/
* Studio: https://studio-pr17748.sandbox.opencraft.hosting/

**Merge deadline**: ASAP

**Testing instructions**:

Each of these steps should succeed.

1. In Studio, Export the DemoX course (or any other large course). 
1. In Studio, Import the DemoX course (or any other large course) into a new course.
1. In LMS, using the Instructor > Data Download area, download any student reports, e.g. grade reports or student profile CSVs.  The downloaded files should be non-empty.
1. In LMS Discussion area, post a large (~1Mb) image.  This should work, and the image should display in the post.

**Reviewers**
